### ISSUE TITLE
Error when uploading files: filename undefined

### ISSUE THREAD

**Issue by EricTheMagician** [projectsend/projectsend#694]
## Type

Please complete with an X between the brackets to indicate the type of submission.
This is a:

[x] Bug report

[] Feature request

### Description of the problem
When I try to upload a file, like .dat file,
I get an error:

File Name
--
undefined

![image](https://user-images.githubusercontent.com/323436/54319291-81303100-45bf-11e9-882f-359db69fd2f2.png)

### List the steps to reproduce the issue

1.Upload a file with extension .dat

### Environment

* ProjectSend version:r1053
* php version: 7.2
* MySQL version:5.7.25

### Additional information

If you add the extension first to the allowed list of extensions, then it works.
I also disabled the extension checking for everyone

### IMPORTANT

Before submitting an issue, please [do a search on the issues page](https://github.com/ignacionelson/ProjectSend/issues) to make sure that the problem hasn't been reported already.

**Comment by ramrajone** [comment]
have you entered the title?  that screenshot is just information only not an error

**Comment by thejinx0r** [comment]
@ramrajone 
How do I do that?
I don't see a location or a field for entering a title.
Here's a bigger screenshot to show that there is no field for me to enter.

![image](https://user-images.githubusercontent.com/323436/55168441-367af100-5149-11e9-8548-815b4a736a75.png)

For clarity, the filename was named  "testfile.sh" but it shows up in the screen shot as "undefined", with no obvious option for entering a new title.

**Comment by kkplein** [comment]
We have seen this kind of behaviour, caused by the "limit upload filetypes" config. Perhaps you should check that you are allowed to upload ".sh" files, even when you have selected NOT to restrict uploaded file types. (yes, that is strange, we feel it's a bug)

**Comment by thejinx0r** [comment]
@kkplein
It does work if I add the .sh files to my allowed file types, which lead to me opening this issue.

However, the file is saved to the filesystem. If it is, I can't find it.
I do see the network being used (vnstat) but I don't see the space on my disk change.

**Comment by kkplein** [comment]
Ah I read your additional information only now. Yes, you discovered the same as us. It's one (of multiple) bugs that still need to be fixed.

Ignacio appears quite silent, but a new version is badly needed, as many of these kinds of bugs seems to be fixed in te development version.
Really HIGH time for a new release, in our opinion.

**Comment by ramrajone** [comment]
could the issue be due to the PHP version?
![image](https://user-images.githubusercontent.com/34359196/55175236-4993be00-5155-11e9-978c-2820686a1a98.png)

**Comment by lmsilva** [comment]
I'm bumping into this same issue and can confirm that uploads only work for extensions defined under "Options \ Security", even though I DISABLED locking down the uploads to specific extensions.

My javascript knowledge is limited BUT I highly suspect the javascript (or swf, and xap?) controls used to handle the uploads are the culprits, as it seems they don't properly POST to upload-process-form.php when the user tries to upload a file with an unknown extension.

When this happens, I noticed that:
- if (isset($_POST['submit'])) { right at the top of the file doesn't execute (because $_POST['submit'] is not properly set)
- and $_POST['finished_files'] just contains an element with "undefined" (when uploading a single file with a "bad extension"

Hope this helps!

**Comment by lmsilva** [comment]
I managed to find and solve the problem.

The problem is in process-upload.php, around line 56:
        $allowedExt = explode(',', $options_values['allowed_file_types'] );
        if (!in_array($fileExt, $allowedExt)) {
            die('{"jsonrpc" : "2.0", "error" : {"code": 104, "message": "Invalid Extension."}, "id" : "id"}');
        };

Basically, this code always gets run, even when extension checks are disabled under "Options \ Security".

In order to fix this, simply replace code with:
// only validate extensions if uploads are blocked by extension!
if ( false === CAN_UPLOAD_ANY_FILE_TYPE ) {
        $allowedExt = explode(',', $options_values['allowed_file_types'] );
        if (!in_array($fileExt, $allowedExt)) {
            die('{"jsonrpc" : "2.0", "error" : {"code": 104, "message": "Invalid Extension."}, "id" : "id"}');
        };
}

This makes it so that extensions only get checked IF extension security is turned on.

**Comment by thejinx0r** [comment]
I just noticed there's a new version (1070) that supposedly fixes this bug. Is that correct? Can this issue be closed now?

**Comment by macinapp2** [comment]
@thejinx0r I just setup a fresh install of Project Send on CentOS 7. I'm having the same issue as above. 
Someone in another post said to chmod 777 the /var/www/html/projectsend/upload/files which did not resolve the issue. 

I replaced the code in process-upload.php with the code from @lmsilva which also did not resolve the issue. I am testing with a .jpg. 

I'm not sure what else to try. 

![Screen Shot 2019-04-26 at 3 52 30 PM](https://user-images.githubusercontent.com/18740461/56833290-14ee5180-683c-11e9-81ea-ad002572e209.png)

**Comment by baibhav-007** [comment]
Hi,
Can you please confirm about the resolution on this problem, as i am facing the same problem..

**Comment by macinapp2** [comment]
> Hi,
> Can you please confirm about the resolution on this problem, as i am facing the same problem..

Hi, I know this isn't the answer you're looking for but I don't believe I ever solved this issue. I instead switched to NextCloud. This was almost a year ago so there may now be a solution. 

Good luck

**Comment by lmsilva** [comment]
> Hi,
> Can you please confirm about the resolution on this problem, as i am facing the same problem..

Check my patch above, as it solves the issue.
I don't think this has ever been properly addressed in the code (seeing as the ticket is still open) but my fix above solves the problem.

**Comment by baibhav-007** [comment]
> > Hi,
> > Can you please confirm about the resolution on this problem, as i am facing the same problem..
> 
> Hi, I know this isn't the answer you're looking for but I don't believe I ever solved this issue. I instead switched to NextCloud. This was almost a year ago so there may now be a solution.
> 
> Good luck

Thanks for the suggestion @macinapp2 , but as i see that nextcloud and Projectsend are different in use. My basic need is that our clients can upload documents and can do self registrations..

But i think this is not possible in nextcloud (Client self registrations)..

**Comment by baibhav-007** [comment]
> > Hi,
> > Can you please confirm about the resolution on this problem, as i am facing the same problem..
> 
> Check my patch above, as it solves the issue.
> I don't think this has ever been properly addressed in the code (seeing as the ticket is still open) but my fix above solves the problem.

Hello,
I am sorry to say that it didn't worked.. Currently i am already using the updated version r1070

**Comment by baibhav-007** [comment]
Hello ,
Good News, as i did solve the problem.. 

All i was to Run Command:
`sudo chmod 777 /var/www/html/projectsend`

Now the files are working well. Just want to know has this led to any security or privacy also..

Please Confirm @ignacionelson @imsilva

**Comment by baibhav-007** [comment]
Hello,

i was able to resolve the issue by running code:
`sudo chmod 777 /var/www/html/projectsend/upload/files`

This is safe...

Please confirm @ignacionelson

**Comment by lmsilva** [comment]
@baibhav-007 it is not as safe as:
chown www. /var/www/html/projectsend/upload/files
chmod 700 /var/www/html/projectsend/upload/files

Basically, I recommend locking the directory to the user running the http server (might be www, httpd or apache depending on your linux distribution).

If you do not, there’s a potential for a race condition attack, during the upload process, by other users running local on the server (e.g. they could create a symlink that lets them overwrite any files the web server user has write access to).

**Comment by alexmaury254** [comment]
I can't figure how to solve this issue! Can someone help with the solution. I have implemented some of the possible solutions in older threads but nothing happens.

**Comment by lmsilva** [comment]
> I can't figure how to solve this issue! Can someone help with the solution. I have implemented some of the possible solutions in older threads but nothing happens.

@alexmaury254 list the permissions of your upload/files/ folder with "ls -ld /var/www/html/projectsend/upload/files/" (or wherever the location of the upload files folder.

**Comment by alexmaury254** [comment]
I'm trying to solve the permissions in Filezilla. But, not really effective. Commands not working. I'm a little confused!

**Comment by lmsilva** [comment]
@alexmaury254 you will probably need privileged shell access so you can fix the permissions.
Basically, a few things are needed:
- either set the folder owner to the user running the web service and set the permissions to 700 (rwx------)
- OR, assuming it is a shared hosting account, set the folder permissions to 777 (rwxrwxrwx)
-- this is not very secure because you'll be giving full access to that folder to everyone on the server BUT, if you do not have full control over the hosting in your server, I don't think you have any other options.

**Comment by ignacionelson** [comment]
This issue is being closed as part of a bulk cleanup. ProjectSend has undergone significant changes since this was reported (including r1720 and r1994), and many older issues no longer apply to the current codebase.

If you are still experiencing this problem on the latest version, please open a new issue with updated details and we'll be happy to look into it. Thank you for your contribution!
